### PR TITLE
Argument serializer

### PR DESCRIPTION
**Pull request by reneklacan**
* Uses @adambird PR as a base (thank you ❤️ )
* Rebases on current master and refactors things a bit
* Makes Rails  `~> 4.0` default version

### CODE REVIEW

**Comment by reneklacan** [comment]
<a href="https:&#x2F;&#x2F;trello.com&#x2F;c&#x2F;v2yhY4Vp&#x2F;562-switch-to-mainstream-resque-mailer"> Switch to Mainstream resque_mailer</a>

**Comment by ErebusBat** [comment]
👍

File: README.md [hunk 1]
```diff
@@ -109,6 +110,15 @@ class ExampleMailer < AsyncMailer
 end
 ```
 
+### Writing an Argument Serializer
+
+By default, the arguments you pass to your mailer are passed as-is to Resque. This
+means you cannot pass things like database-backed objects. If you'd like to write
+your own serializer to do enable such things, simply write a class that implements
```
Comment (philippt) [inline README.md]: "to do enable" is too much

**Comment by philippt** [comment]
:+1:

**Pull request merged**